README.md

```diff
@@ -2,20 +2,175 @@
 
 > Test client for Xo-Server
 
-## Usage
+## Setup
+
+- Xo-server needs to be installed on `localhost:9000`.  
+Otherwise, you should modifiy the `xoServerUrl` in the `congig.toml`.
+
+- You should have a `admin@admin.net` account with password `admin`.
+
+- There should be a Xen Server available at `192.168.1.3` with the credentials `root` and password `qwerty`.
 
-xo-server needs to be installed on `localhost:9000`.
-You should also have a `admin@admin.net` account with password `admin`.
+## Adding a test
 
-There should be a Xen Server available at `192.168.1.3`with the credentials `root` and password `qwerty`.
+### Organization
 
-Warning ! This client will create and delete a lot of items (users, groups, servers ...) to test features
+Reminder: the xo commands are composed of a **XO object** and a *method* (**user**.*create*, **job**.*getAll*, **vm**.*delete*,...). 
 
 ```
-> npm ci
-> npm test
+src
+├─ user
+|     ├─ __snapshots__
+|     |     └─ user.spec.js.snap 
+|     └─ user.spec.js
+├─ job
+¦     └─ job.spec.js
+¦
+¦
+├─ scenarios
+|     ├─ __snapshots__
+¦     ¦     ¦
+¦     ¦     ├─ revertVm.spec.js.snap 
+¦     ¦     └─ migrateVm.spec.js.snap
+|     ├─ revertVm.spec.js
+|     └─ migrateVm.spec.js
+├─ _xoWithTestHelpers.js
+└─ util.js
 ```
 
+There are two ways to classify tests:
+
+  + By XO objects.  
+    In each file `[XO object].spec.js`, the tests must be grouped by methods.  
+    
+    Let's take an example with the XO object **user** and some of its methods:  
+    ```
+    describe("user", () => {
+      describe(".create() :", () => {
+        it("creates a user with permission", async () => {
+          /* some tests */
+        });
+        it("creates a user without permission", async () => {
+          /* some tests */
+        });
+      });
+
+      describe(".delete() :", () => {
+        it("deletes a user successfully with id", async () => {
+          /* some tests */
+        });
+      });
+    });
+    ```
+    The test must be simple : one feature by test.  
+
+  + By scenarios.  
+    In the folder `scenarios`: name the test files with what they do *mainly*.  
+    Be as precise as possible in the title of the test.  
+    
+    For example, in the `revertVm.spec.js` file:  
+    ```
+    describe("revertVm :", {
+      it("fails trying to execute three times a rolling snapshot with 2 as retention and reverts the VM to the first snapshot which is normally removed", async () => {
+        /* some tests */
+      });
+    });
+    ```
+    
+### Test environment
+
+- The test environment must remain the same before and after each test:  
+  * each resource created must be deleted  
+  * no existing resource before running the tests should be deleted or modified.  
```

Comment:
1 - It talks about the resources created by the user  which must be deleted after the tests.
2 - It only talks about the existing resources before tests must not be modified or deleted.

Maybe `The test environment must remain the same before and after each test` is enough? I just wanted to be sure the developers understand  what I meant. I can insert an example and remove these two points.

Reply:
You can keep it i just not understand the difference because it have a typo. `should be` -> `should not be`. or simply `existing resources should not be altered`.